### ISSUE TITLE
Error code 256

### ISSUE THREAD

**Issue by The66thDopefish** [eskay993/gamefiles#15]
**Details:**

- Which game are you trying to install?
SP Football Life 2024

- What platform/distro are you trying it on (e.g. Steam Deck)?
Steam Deck

**Issue Description:**
I have been banging my head against the wall trying to troubleshoot this myself to no avail.  I have followed the instructions as given and at step 9 I get this error.  I have tried clearing the Lutris cache with no change.

**Error Log:**

Started initial process 78 from /home/deck/.var/app/net.lutris.Lutris/data/lutris/runtime/winetricks/winetricks --unattended remove_mono win7
Start monitoring process.
------------------------------------------------------
warning: Your version of wine GE-Proton8-15 is no longer supported upstream. You should upgrade to 7.x
------------------------------------------------------
------------------------------------------------------
WINEPREFIX INFO:
Drive C: total 384
drwxrwxrwx 1 deck deck 131072 Oct  4 11:11 .
drwxrwxrwx 1 deck deck 131072 Oct  4 11:22 ..
drwxrwxrwx 1 deck deck 131072 Oct  4 11:11 windows

Registry info:
/run/media/deck/3130-3066/system.reg:#arch=win64
/run/media/deck/3130-3066/user.reg:#arch=win64
/run/media/deck/3130-3066/userdef.reg:#arch=win64
------------------------------------------------------
------------------------------------------------------
warning: /home/deck/.var/app/net.lutris.Lutris/data/lutris/runners/wine/lutris-GE-Proton8-15-x86_64/bin/wine cmd.exe /c echo '%AppData%' returned empty string, error message "wine: could not load kernel32.dll, status c0000135" 
------------------------------------------------------
Monitored process exited.
Initial process has exited (return code: 256)
Exit with return code 256

**Comment by The66thDopefish** [comment]
A couple of things I will mention:

- I was initially trying to install SP Football Life 2024 on an external SD card, and I think perhaps that was fouling things up? I am now trying to install straight to the Deck storage, and that seems to be working better as I have gotten to step 10 (running the SP installer).
- I also deleted the Wine files (but not the Lutris files) before trying again.  I can't say if that's made the difference as much as changing the installation location.

So far it looks like I'm having success.  I don't know enough about Linux to understand why Lutris wouldn't be able to install to the external SD card.

**Comment by eskay993** [comment]
Hi

The error in your first post is usually because Lutris didn't download some components correctly. Clearing Lutris cache usually fixes those. Follow these steps to clear Lutris cache:

https://github.com/eskay993/gamefiles/tree/main/sp-football-life-2024#how-to-clear-your-lutris-cache

If you get an error message when you first launch Lutris after doing this, then try clearing cache again. I was having issues with Lutris a few days ago so might be something on their end.

Sounds like you already read through this - https://github.com/eskay993/gamefiles/issues/4

Just to confirm:

- What did you put in Flatseal to give Lutris access to the SD card? Should be something like `/run/media:rw'` in the Lutris section under Volumes.

- What path are you installing the game to on your SD card? Should be something like `/run/media/mmcblk0p1/Games/sp-football-life-2024`. You should this in Step 6 of the guide.

- In your Edit 4, when you say it hangs, does it hang with the error in the first post? or something else?

Note: where the yaml file is located should not matter. As long as Lutris can access that location, so your Downloads folder is fine.

Cheers

**Comment by The66thDopefish** [comment]
- How often do I have to clear the Lutris cache? Before every installation attempt?
- How about deleting/clearing the Wine files? e.g. `sudo rm -rf /var/lib/wine` `sudo rm -rf /var/wine` `sudo rm -rf /wine` 'rm rf ~/.config/wine'
- Is the extra apostrophe in your code `/run/media:rw` an error? Anyway I have had a line without the ':rw' but also another line for the direct address of the folder I was trying to install to (see below).  Seems like it would be redundant and that '/run/media:rw' would cover the entire SD card.
- I did read issue #4 and, regarding Step 6, when I try to install to `/run/media/mmcblk0p1/` I get the message "**Warning** The destination folder is not writable by the current user." and the Wine prefix installation doesn't even begin.  So I have another folder (`/run/media/deck/3130-3066/Games/sp-football-life-2024`) into which I am trying to install the game.
- When I say the Wine prefix creation hangs, I mean it simply does not proceed beyond a window saying "The Wine configuration in /run/media/deck/3130-3066/Games/sp-football-life-2024 is being updated, please wait..." Yesterday I simply let it sit there and a couple of hours later this window was still open and the "Installing Football Life 2024" window showed me the error in my OP.

**Comment by eskay993** [comment]
No, do not delete those wine folders!! They are nothing to do with Lutris. Those are system wine folders.  General rule, don't randomly delete files unless you really know what for sure what they do! :)

Yes the extra apostrophe was en error. It should be as you wrote it `/run/media:rw`. Apologies.

I'm confused why your SD card is mounted to `/run/media/deck/3130-3066`. Are you using an SD card plugged into the deck? Or an SD card plugged into a USB dongle of some sort?

SD card plugged into the Deck should mount to `/run/media/mmcblk0p1`. 

If you are using an SD card plugged into a USB dongle, are you sure the card and the dongle work, and the card is formatted correctly? If you are using this setup, then there are a lot of variables so would be hard to pin point.

**Comment by The66thDopefish** [comment]
Okay, I won't bother with the Wine stuff again :D

So my SD card (which is plugged straight into the Deck) also contains save data from my Switch.  I don't know if that would affect this at all? Regardless I do see the `mmcblk0p1` directory, but when I go to create the folders suggested in Dolphin I am told that it is not writable by the user.  So I have been using the `3130-3066` folder which is where the Nintendo files are located.

I'll reiterate that at one point I didn't have any issues getting the Wine prefix configured so I don't know if the destination is that critical?

What I may do (if all else fails) is save the Switch contents on the SD card to my computer, format the card and start fresh.

**Comment by eskay993** [comment]
Ah right. That would explain it. The Deck is detecting it as an external drive/SD card and mounting it in that folder you mentioned, and not in the default location for the internal SD card. 

I wouldn't recommend using the same card for your Switch and Deck. It's playing with fire! I suggest you backup your Switch saves as you suggest and use this card exclusively for Deck. Let the Deck format it properly for internal use. It will then mount it in mmcblk0p1 and will be writable. 

Then install the game there and test. I am releasing a new version of the script shortly to address an unrelated issue so try that version. I'll put it live in the next 10 mins or so.

**Comment by The66thDopefish** [comment]
Okay, I was already in the process of transferring the Switch data off of this SD card, then I will format the card and give it another shot.  I will let you know how it goes!

**Comment by The66thDopefish** [comment]
Looks like, barring any last minute surprises, that was the solution.  I went to format the card and quickly realized that it was not in a format that worked well with the Steam Deck.  I went to try to format it in KDE Partition Manager and it wouldn't let me format it there.  Fortunately SteamOS in Settings has its own Format SD option and that appeared to do correct the problem.  Everything has worked without a hitch thus far, and SP Football Life 2024 installation is going smoothly, if slowly.  Unless there's another issue I can't troubleshoot, I'm gonna close this one.  Thanks for your help essay993!
